File: ios/NotificationManager.m
//
//  NotificationManager.m
//  COAktau
//
//  Created by Mac on 03.09.18.
//  Copyright © 2018 Facebook. All rights reserved.
//

#import "NotificationManager.h"
#import <Foundation/Foundation.h>
#import <UIKit/UIKit.h>

@import PushNotifications;

@implementation NotificationManager

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(enableNotification:(NSString *)lang) {
  NSError *error;
  if([lang isEqualToString:@"rus"]) {
    [[PushNotifications shared] subscribeWithInterest:@"rus" error:&error completion:^{}];
  } else {
    [[PushNotifications shared] subscribeWithInterest:@"kaz" error:&error completion:^{}];
  }
}

RCT_EXPORT_METHOD(disableNotification) {
  NSError *error;
  [[PushNotifications shared] unsubscribeWithInterest:@"rus" error:&error completion:^{}];
  [[PushNotifications shared] unsubscribeWithInterest:@"kaz" error:&error completion:^{}];
}

RCT_EXPORT_METHOD(subscribeRus) {
  NSError *error;
  [[PushNotifications shared] unsubscribeWithInterest:@"kaz" error:&error completion:^{}];
  [[PushNotifications shared] subscribeWithInterest:@"rus" error:&error completion:^{}];
}

RCT_EXPORT_METHOD(subscribeKaz) {
  NSError *error;
  [[PushNotifications shared] unsubscribeWithInterest:@"rus" error:&error completion:^{}];
  [[PushNotifications shared] subscribeWithInterest:@"kaz" error:&error completion:^{}];
}

@end

File: android/app/src/main/java/com/coaktau/NotificationManager.java
package com.coaktau;

import android.util.Log;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.pusher.pushnotifications.PushNotifications;

public class NotificationManager extends ReactContextBaseJavaModule {

    public NotificationManager(ReactApplicationContext reactContext) {
        super(reactContext);
    }

    @Override
    public String getName() {
        return "NotificationManager";
    }

    @ReactMethod
    public void enableNotification(String lang) {
        if(lang.equals("rus")) {
            PushNotifications.subscribe("rus");
        } else {
            PushNotifications.subscribe("kaz");
        }
    }

    @ReactMethod
    public void disableNotification() {
        PushNotifications.unsubscribe("rus");
        PushNotifications.unsubscribe("kaz");
    }

    @ReactMethod
    public void subscribeRus() {
        PushNotifications.unsubscribe("kaz");
        PushNotifications.subscribe("rus");
        Log.v("rus", "msg");
    }

    @ReactMethod
    public void subscribeKaz() {
        PushNotifications.unsubscribe("rus");
        PushNotifications.subscribe("kaz");
        Log.v("kaz", "msg");
    }
}